src/coins.h

```diff
@@ -110,7 +110,12 @@ class SaltedOutpointHasher
 public:
     SaltedOutpointHasher();
 
-    uint64_t operator()(const COutPoint &id) const { return SipHashUint256Extra(k0, k1, id.hash, id.n); }
+    /**
+     * This *must* return size_t. With Boost 1.46 on 32-bit systems the
+     * unordered_map will behave unpredictably if the custom hasher returns a
+     * uint64_t, resulting in failures when syncing the chain (#4634).
+     */
+    size_t operator()(const COutPoint &id) const { return SipHashUint256Extra(k0, k1, id.hash, id.n); }
```

Comment:
I think you can leave this as it was, this return was changed to uint64_t some time ago because we don't use boost unordered map anymore and no longer have the issue to worry about.

Reply:
ok